internal/serialization/field_operations.go

```diff
@@ -0,0 +1,373 @@
+	return variableKindSize
+}
+
+type TimestampWithTimezoneFieldKindOperation struct{}
+
+func (TimestampWithTimezoneFieldKindOperation) KindSizeInBytes() int32 {
+	return variableKindSize
+}
+
+type ArrayOfTimestampWithTimezoneFieldKindOperation struct{}
+
+func (ArrayOfTimestampWithTimezoneFieldKindOperation) KindSizeInBytes() int32 {
+	return variableKindSize
+}
+
+type CompactFieldKindOperation struct{}
+
+func (CompactFieldKindOperation) KindSizeInBytes() int32 {
+	return variableKindSize
+}
+
+type ArrayOfCompactFieldKindOperation struct{}
+
+func (ArrayOfCompactFieldKindOperation) KindSizeInBytes() int32 {
+	return variableKindSize
+}
+
+type NullableBooleanFieldKindOperation struct{}
+
+func (NullableBooleanFieldKindOperation) KindSizeInBytes() int32 {
+	return variableKindSize
+}
+
+type ArrayOfNullableBooleanFieldKindOperation struct{}
+
+func (ArrayOfNullableBooleanFieldKindOperation) KindSizeInBytes() int32 {
+	return variableKindSize
+}
+
+type NullableInt8FieldKindOperation struct{}
+
+func (NullableInt8FieldKindOperation) KindSizeInBytes() int32 {
+	return variableKindSize
+}
+
+type ArrayOfNullableInt8FieldKindOperation struct{}
+
+func (ArrayOfNullableInt8FieldKindOperation) KindSizeInBytes() int32 {
+	return variableKindSize
+}
+
+type NullableInt16FieldKindOperation struct{}
+
+func (NullableInt16FieldKindOperation) KindSizeInBytes() int32 {
+	return variableKindSize
+}
+
+type ArrayOfNullableInt16FieldKindOperation struct{}
+
+func (ArrayOfNullableInt16FieldKindOperation) KindSizeInBytes() int32 {
+	return variableKindSize
+}
+
+type NullableInt32FieldKindOperation struct{}
+
+func (NullableInt32FieldKindOperation) KindSizeInBytes() int32 {
+	return variableKindSize
+}
+
+type ArrayOfNullableInt32FieldKindOperation struct{}
+
+func (ArrayOfNullableInt32FieldKindOperation) KindSizeInBytes() int32 {
+	return variableKindSize
+}
+
+type NullableInt64FieldKindOperation struct{}
+
+func (NullableInt64FieldKindOperation) KindSizeInBytes() int32 {
+	return variableKindSize
+}
+
+type ArrayOfNullableInt64FieldKindOperation struct{}
+
+func (ArrayOfNullableInt64FieldKindOperation) KindSizeInBytes() int32 {
+	return variableKindSize
+}
+
+type NullableFloat32FieldKindOperation struct{}
+
+func (NullableFloat32FieldKindOperation) KindSizeInBytes() int32 {
+	return variableKindSize
+}
+
+type ArrayOfNullableFloat32FieldKindOperation struct{}
+
+func (ArrayOfNullableFloat32FieldKindOperation) KindSizeInBytes() int32 {
+	return variableKindSize
+}
+
+type NullableFloat64FieldKindOperation struct{}
+
+func (NullableFloat64FieldKindOperation) KindSizeInBytes() int32 {
+	return variableKindSize
+}
+
+type ArrayOfNullableFloat64FieldKindOperation struct{}
+
+func (ArrayOfNullableFloat64FieldKindOperation) KindSizeInBytes() int32 {
+	return variableKindSize
+}
+
+func FieldOperations(fieldKind pserialization.FieldKind) FieldKindOperation {
+	switch fieldKind {
+	case pserialization.FieldKindBoolean:
+		return BooleanFieldKindOperation{}
```

Comment:
minor: maybe we can put these into an array for much faster lookups

Reply:
That makes sense.